learning/katas/go/Core Transforms/CoGroupByKey/CoGroupByKey/pkg/task/task.go

```diff
@@ -0,0 +1,52 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"fmt"
+	"github.com/apache/beam/sdks/go/pkg/beam"
+)
+
+func ApplyTransform(s beam.Scope, fruits beam.PCollection, countries beam.PCollection) beam.PCollection {
+	fruitsKV := beam.ParDo(s, func(e string) (string, string) {
+		return string(e[0]), e
+	}, fruits)
+
+	countriesKV := beam.ParDo(s, func(e string) (string, string) {
+		return string(e[0]), e
+	}, countries)
+
+	grouped := beam.CoGroupByKey(s, fruitsKV, countriesKV)
+	return beam.ParDo(s, func(key string, f func(*string) bool, c func(*string) bool, emit func(string)) {
```

Comment:
I understand the convention of Go that prefers shorter variable name, but I personally find that `f` and `c` are cryptic.
But if it's well understood, I guess I'm just too old for this :)

Btw, I like the variable naming in the examples at the top of the godoc.
https://godoc.org/github.com/apache/beam/sdks/go/pkg/beam

![image](https://user-images.githubusercontent.com/5459430/82823012-58047380-9ed9-11ea-896b-79ef8789ebe5.png)

Reply:
I agree with you and realizing for code readability purposes it makes sense.  I created [BEAM-10091](https://issues.apache.org/jira/browse/BEAM-10091) when we are all complete with the series of Go SDK katas.  I'd like to perform an overall cleanup for naming consistency and code readability.

Reply:
+1 to using unambiguous names here. 
In this case it's especially important to disambiguate between the iterator parameters (currently f and c) and the emiter parameter since they're both function parameters.
eg, using iter1, iter2 would be better than unrelated single character names. Also +1 to doing a global cleanup instead of immediately back updating everything previously. That said, there's no harm in starting in *this* PR.

Reply:
Thanks for creating the cleanup task!